src/ingest_validation_tools/validation_utils.py

```diff
@@ -30,96 +34,167 @@ def dict_reader_wrapper(path, encoding: str) -> list:
+            )
+        elif "orcid_id" in rows[0]:
+            message += (
+                ' Has "orcid_id": Contributors TSV found where metadata TSV expected.'
+            )
+        else:
+            message += f' Column headers in TSV: {", ".join(rows[0].keys())}'
+        raise PreflightError(message)
+    return SchemaVersion(
+        assay_type_data["assaytype"].lower(),
+        directory_path=directory_path,
+        path=path,
+        raw_rows=rows,
+        soft_assay_data=assay_type_data,
     )
 
 
-def get_directory_schema_versions(
-    tsv_path: str, schema_version: str, encoding: str
-) -> list:
-    parent = Path(tsv_path).parent
-    data_paths = [r.get("data_path") for r in _read_rows(tsv_path, encoding)]
-    return list(
-        set(
-            _get_directory_schema_version(parent / path, schema_version)
-            for path in data_paths
-            if path
+def get_other_schema_name(rows: List, path: str) -> Optional[str]:
+    other_types = {
+        "organ": ["organ_id"],
+        "sample": ["sample_id"],
+        "sample-block": ["sample_id"],
+        "sample-suspension": ["sample_id"],
+        "sample-section": ["sample_id"],
+        "contributors": ["orcid", "orcid_id"],
+        "antibodies": ["antibody_rrid", "antibody_name"],
+    }
+    other_type: DefaultDict[str, list] = defaultdict(list)
+    for field in rows[0].keys():
+        if field == "sample_id":
+            if rows[0].get("type"):
+                other_type.update({f"sample-{rows[0]['type']}": ["sample_id"]})
+            else:
+                other_type.update({"sample": ["sample_id"]})
+        else:
+            match = {key: field for key, value in other_types.items() if field in value}
+            other_type.update(match)
+    if other_type and (
+        "assay_name" in rows[0].keys() or "dataset_type" in rows[0].keys()
+    ):
+        raise PreflightError(
+            f"Metadata TSV contains invalid field: {list(other_type.values())}"
         )
-    )
+    if len(other_type) == 1:
+        return list(other_type.keys())[0]
+    elif len(other_type) > 1:
+        raise PreflightError(
+            f"Multiple types found for path {path} based on fields {list(other_type.values())}"
+        )
+    else:
+        return None
+
+
+def get_ingest_api_env(env: str) -> str:
+    if env in ["dev", "test", "stage"]:
+        return f"https://ingest-api.{env}.hubmapconsortium.org/assaytype"
+    elif env == "prod":
+        return "https://ingest.api.hubmapconsortium.org/assaytype"
+    elif env == "local":
+        return "http://localhost:5000/assaytype"
+    else:
+        raise Exception(f"Environment {env} not found!")
+
 
+def get_assaytype_data(
+    row: Dict,
+    globus_token: str,
+    path: Path,
+    env: str = "dev",
+    offline: bool = False,
+) -> Dict:
+    if "version" in row.keys():
+        row["version"] = int(row["version"])
+    if offline or not globus_token:
+        return mock_response(path, row)
```

Comment:
We should rethink this for future iterations. Put a TODO here so that we can discuss this once this push is over.

Reply:
Done!